### PR TITLE
Fix edit info on grant page

### PR DESCRIPTION
**Pull request by PierrickGT**
##### Description
As a grant owner, I wish to be able to edit infos of my grant.
This PR fix a bug who made it impossible to edit infos of a grant if you where not doing it from the first tab, it was due to the Quill editor not being conditioned to be loaded only on the first tab.

##### Screencast
Here is a recording of the fix
fix screencast

##### Refers/Fixes
Bounty: https://gitcoin.co/issue/gitcoinco/web/4786/3227
Issue: https://github.com/gitcoinco/web/issues/4786

### CODE REVIEW

**Comment by PierrickGT** [comment]
> Could you add a recording of the fix + update the GitHub description so other folks know exactly what is done before they check out the code ?
> 
> Code looks alright ! Will pull it in and check it out locally ^_^

I've added a better description of the issue and a gif recording of the fix.

File: app/assets/v2/js/grants/detail.js [hunk 1]
```diff
@@ -86,24 +91,35 @@ $(document).ready(function() {
 
     let edit_title = $('#form--input__title').val();
     let edit_reference_url = $('#form--input__reference-url').val();
-    let edit_admin_profile = $('#grant-admin option').last().text();
-    let edit_description = grant_description.getText();
-    let edit_description_rich = JSON.stringify(grant_description.getContents());
+    let edit_admin_profile = $('#grant-admin option')
+      .last()
+      .text();
```
Comment (thelostone-mc) [inline app/assets/v2/js/grants/detail.js]: ^ could we throw those two into L94 as opposed to breaking it ?

Comment (PierrickGT) [reply app/assets/v2/js/grants/detail.js]: Done: https://github.com/gitcoinco/web/pull/4793/commits/a5a190018ba89f013f0d16880820a89a85b706eb
Must have been formatted by my Prettier plugin

**Comment by PierrickGT** [comment]
Thanks for reviewing my PR @thelostone-mc !
Do you know how I could fix the coverage errors?

**Comment by thelostone-mc** [comment]
@PierrickGT Ah i guess that's alright for now! we are yet to wire in a testing library for frontend

**Comment by PierrickGT** [comment]
@thelostone-mc ok thanks! Waiting for my work to be approved on Gitcoin before sending this PR and then merging it.

**Comment by thelostone-mc** [comment]
@PierrickGT you should be able to apply for work on it ^_^
Could you click on start work and I'll go ahead and approve you

**Comment by thelostone-mc** [comment]
@PierrickGT looks like we have CI failures :| could you resolve that ?

**Comment by PierrickGT** [comment]
> @PierrickGT looks like we have CI failures :| could you resolve that ?

I've fixed it, there was a formatting error introduced by merging master.

**Comment by PierrickGT** [comment]
> Hey @PierrickGT - looks like there were a lot of unrelated style changes in your PR here, can you revert those so its easier to review? Generally we keep the changes limited to the scope of the issue. Thanks!

I've fixed it, except for some indentations that were not right: https://github.com/gitcoinco/web/pull/4793/commits/162ee5602958ef32544512bd870795407ca02491
You should definitely use Prettier or even eslint in order to have a more readable JS codebase.

**Comment by PierrickGT** [comment]
> @PierrickGT we've got precommit hooks which run eslint + stylelint.
> It's documented in our [Contributing.md](https://github.com/gitcoinco/web/blob/be9b70181dd87be6a5ff6e4cf64a832b1c8ca9ee/docs/CONTRIBUTING.md#prerequisites) file
> 
> Did you check that out ?

Oh I didn't know about it. I've set it up so we'll see how it works out for the future commits.

File: app/assets/v2/js/grants/detail.js [hunk 2]
```diff
@@ -176,36 +178,41 @@ $(document).ready(function() {
...
+            data: {
+              'contract_owner_address': contract_owner_address
+            },
+            success: function(json) {
+              window.location.reload(false);
+            },
+            error: function() {
+              _alert({ message: gettext('Changing the contract owner address failed to save. Please try again.') }, 'error');
+            }
+          });
```
Comment (thelostone-mc) [inline app/assets/v2/js/grants/detail.js]: @PierrickGT could you undo the linting you've done here ? 
It makes it confusing for reviewers to see what exactly your changes are

Comment (PierrickGT) [reply app/assets/v2/js/grants/detail.js]: I've fixed it: https://github.com/gitcoinco/web/pull/4793/commits/3a57279e3300b250ac2a80e1bc236b9302c8b039

**Comment by danlipert** [comment]
Hey @PierrickGT - I just tested this before our production deploy and its broken - the quill library is not loading correctly when I view a grant somebody else created - see screenshot: 
Screen Shot 2019-08-01 at 12 25 45 AM

please fix and resubmit

**Comment by PierrickGT** [comment]
> Hey @PierrickGT - I just tested this before our production deploy and its broken - the quill library is not loading correctly when I view a grant somebody else created - see screenshot:
> Screen Shot 2019-08-01 at 12 25 45 AM
> 
> please fix and resubmit

My bad, I'm gonna take a look at it and fix it.

**Comment by PierrickGT** [comment]
I've reopened a PR where I've fixed the condition to initialize Quill in the file `app/grants/templates/grants/detail/tabs.html`, I've forgot about the case where you don't have `?tab=description` in the url.

https://github.com/gitcoinco/web/pull/4928

**Comment by octavioamu** [comment]
@PierrickGT is that PR with this PR changes also? because we had reverted this merge since was failing in our deploy tests.

**Comment by PierrickGT** [comment]
> @PierrickGT is that PR with this PR changes also? because we had reverted this merge since was failing in our deploy tests.

Yes, it's the same PR but with the fix.

**Pull request merged**